### PR TITLE
Supported ActiveMQ authentication

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-tram-consumer-activemq/src/main/java/io/eventuate/tram/consumer/activemq/MessageConsumerActiveMQImpl.java [hunk 1]
```diff
@@ -42,12 +42,27 @@
   private AtomicBoolean runFlag = new AtomicBoolean(true);
 
   public MessageConsumerActiveMQImpl(String url) {
-    this(url, Collections.emptyMap());
+    this(url, Collections.emptyMap(), Optional.empty(), Optional.empty());
   }
 
-  public MessageConsumerActiveMQImpl(String url, Map<String, ChannelType> messageModes) {
+  public MessageConsumerActiveMQImpl(String url,
```
Comment (cer) [inline eventuate-tram-consumer-activemq/src/main/java/io/eventuate/tram/consumer/activemq/MessageConsumerActiveMQImpl.java]: Why have more than one constructor? It makes the class confusing. If test code needs multiple constructors then define in src/test a class with factory methods

File: eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java [hunk 2]
```diff
@@ -17,7 +19,11 @@ public PublishingFilter activeMQDuplicatePublishingDetector() {
   }
 
   @Bean
-  public DataProducerFactory activeMQDataProducerFactory(@Value("${activemq.url}") String activeMQURL) {
-    return () -> new EventuateActiveMQProducer(activeMQURL);
+  public DataProducerFactory activeMQDataProducerFactory(@Value("${activemq.url}") String activeMQURL,
```
Comment (cer) [inline eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java]: @Value("${activemq.url}") seems to occur multiple times. why?

Comment (dartartem) [reply eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java]: I do not see duplicates, one line with it is removed and other line is added.

File: eventuate-tram-consumer-activemq/src/main/java/io/eventuate/tram/consumer/activemq/MessageConsumerActiveMQImpl.java (hunk 1 above)
Comment (dartartem) [reply eventuate-tram-consumer-activemq/src/main/java/io/eventuate/tram/consumer/activemq/MessageConsumerActiveMQImpl.java]: I will fix

File: eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java (hunk 2 above)
Comment (cer) [reply eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java]: Sorry. I was referring to the user id/password: `@Value("${activemq.user:#{null}}") `

Comment (dartartem) [reply eventuate-tram-cdc-mysql-connector/src/main/java/io/eventuate/tram/cdc/mysql/connector/configuration/ActiveMQMessageTableChangesToDestinationsConfiguration.java]: url, userid, user password used in several place.
I want to create a separate property class in the new projects (separate cdc).
Probably it should be done in common application.

**Pull request merged**